traffic_control/tests/test_barrier_api.py

```diff
@@ -1,11 +1,37 @@
 import datetime
 
+import pytest
 from django.urls import reverse
 from rest_framework import status
 
 from traffic_control.models import BarrierPlan, BarrierReal, ConnectionType, Reflective
 
-from .test_base_api import TrafficControlAPIBaseTestCase
+from .factories import get_api_client, get_barrier_plan, get_barrier_real
+from .test_base_api import (
+    line_location_test_data,
+    point_location_test_data,
+    TrafficControlAPIBaseTestCase,
+)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "location,query_location,expected",
+    [*point_location_test_data, *line_location_test_data],
+)
+def test_filter_barrier_plans_location(location, query_location, expected):
```

Comment:
About code readability in `{"location": query_location.ewkt}`. It might be better to use `location_query` or similar to make code more readable.

Reply:
Changed as requested.